src/ts/localization/i18n.ts

```diff
@@ -0,0 +1,173 @@
+import vocabularyDe from './languages/de.json';
+import vocabularyEn from './languages/en.json';
+import { LocalizationConfig } from '../uimanager.js';
+
+export const defaultTranslations: BitmovinPlayerUiTranslations = {
+  'en': vocabularyEn,
+  'de': vocabularyDe,
+};
+
+export const defaultLocalizationConfig: LocalizationConfig = {
+  language: 'en',
+  translations: defaultTranslations,
+};
+
+type LocalizableCallback = () => string;
+export type LocalizableText = string | LocalizableCallback;
+
+interface Vocabulary {
+  'settings.video.quality': string;
+  'settings.audio.quality': string;
+  'settings.audio.track': string;
+  'settings.audio.mute': string;
+  'settings.subtitles': string;
+  'settings.subtitles.font.color': string;
+  'settings.subtitles.font.opacity': string;
+  'settings.subtitles.background.color': string;
+  'settings.subtitles.background.opacity': string;
+  'settings.subtitles.font.size': string;
+  'settings.subtitles.characterEdge': string;
+  'settings.subtitles.characterEdge.raised': string;
+  'settings.subtitles.characterEdge.depressed': string;
+  'settings.subtitles.characterEdge.uniform': string;
+  'settings.subtitles.characterEdge.dropshadowed': string;
+  'settings.subtitles.font.family': string;
+  'settings.subtitles.font.family.monospacedserif': string;
+  'settings.subtitles.font.family.proportionalserif': string;
+  'settings.subtitles.font.family.monospacedsansserif': string;
+  'settings.subtitles.font.family.proportionalsansserif': string;
+  'settings.subtitles.font.family.casual': string;
+  'settings.subtitles.font.family.cursive': string;
+  'settings.subtitles.font.family.smallcapital': string;
+  'settings.subtitles.window.color': string;
+  'settings.subtitles.window.opacity': string;
+  'colors.white': string;
+  'colors.black': string;
+  'colors.red': string;
+  'colors.green': string;
+  'colors.blue': string;
+  'colors.cyan': string;
+  'colors.yellow': string;
+  'colors.magenta': string;
+  'percent': string;
+  'settings': string;
+  'ads.remainingTime': string;
+  'pictureInPicture': string;
+  'appleAirplay': string;
+  'googleCast': string;
+  'vr': string;
+  'off': string;
+  'auto': string;
+  'back': string;
+  'reset': string;
+  'replay': string;
+  'normal': string;
+  'default': string;
+  'open': string;
+  'fullscreen': string;
+  'speed': string;
+  'playPause': string;
+}
+
+export type CustomVocabulary<V> = V & Partial<Vocabulary>;
+
+export interface BitmovinPlayerUiTranslations {
+  [key: string]: CustomVocabulary<Record<string, string>>;
+}
+
+class I18n {
+  private language: string;
+  private vocabulary: CustomVocabulary<Record<string, string>>;
+
+  constructor(config: LocalizationConfig) {
+    this.setConfig(config);
+  }
+
+  public setConfig(config: LocalizationConfig) {
+    const browserLanguageDetection = config.browserLanguageDetection != null ? config.browserLanguageDetection : true;
```

Comment:
```ts
config = {
  browserLanguageDetection: true,
  ...config,
};
```

Reply:
imo. we should not overwrite fn. parameter, even if it makes it easier for our current situation

Reply:
This has zero side effects, but if you want to play save then do it like this:

```ts
const mergedConfig = {
  browserLanguageDetection: true,
  ...config,
};
```

Reply:
yes it can be done but why do we need to merge it like that? I am basically assigning a default value to variable if it is not user defined. not sure what the downside of  ` ? : ` is

Reply:
It is not pretty and difficult to read.

Reply:
Actually `browserLanguageDetection: true` should be moved to `defaultLocalizationConfig` and then:

```ts
const mergedConfig = {
  ...defaultLocalizationConfig,
  ...config,
};
```